src/inject.cc

```diff
@@ -64,12 +66,15 @@ bool InjectDll(const uint32_t ProcessId, const std::filesystem::path &Path) {
 
   if (ExitCode == 0) {
     printf("/!\\ The thread failed to load the dll.\n");
+    return EXIT_FAILURE;
```

Comment:
Got it. I guess I was also missing the cleaning up in case of virtual memory allocation/ thread creation failure. 82f9852b89fce289ef7896c032b5f4b74dc5cb35 tries to fix all of this.